### PR TITLE
💎 Gemma 3 SFT example on Codeforces dataset

### PR DESCRIPTION
**Pull request by qgallouedec**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/trl/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a GitHub issue? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/trl/tree/main/docs).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by burtenshaw** [comment]
@qgallouedec Amazing work. thanks! a few questions:

- Is padding like this necessary for all trainers?
- do we need to patch other tokens like this:

```python
processor.pad_token_id = processor.tokenizer.pad_token_id
processor.bos_token_id = processor.tokenizer.bos_token_id
processor.eos_token_id = processor.tokenizer.eos_token_id
```

- did you see how unsloth is dealing with missing token id's: https://huggingface.co/unsloth/gemma-3-4b-it/commit/90fe72f525abc73ff7283c23e6ceccea5d4273bb . Do you think we should open a PR for changes on the hub repo?

**Comment by qgallouedec** [comment]
> Is padding like this necessary for all trainers?

usually tokenizers have a pad method. Here, the gemma processor doesn't. But maybe we shouldn't use the processor and directly load the tokenizer? Checking

File: examples/scripts/sft_gemma3.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+if __name__ == "__main__":
+    # Load dataset
+    train_dataset = load_dataset("open-r1/codeforces-cots", split="train[:100]")
+    train_dataset = train_dataset.remove_columns("prompt")
+
+    # Load model and processor
+    model_id = "google/gemma-3-4b-it"
+    processor = AutoProcessor.from_pretrained(model_id)
+    model = AutoModelForImageTextToText.from_pretrained(model_id, attn_implementation="eager")
```
Comment (kashif) [inline examples/scripts/sft_gemma3.py:31]:
```suggestion
    model = AutoModelForImageTextToText.from_pretrained(model_id, attn_implementation="sdpa")
```

Comment (qgallouedec) [reply examples/scripts/sft_gemma3.py:31]: @kashif I get this warning:

```
It is strongly recommended to train Gemma3 models with the `eager` attention implementation instead of `sdpa`. Use `eager` with `AutoModelForCausalLM.from_pretrained('<path-to-checkpoint>', attn_implementation='eager')`.
```

File: trl/trainer/sft_trainer.py [hunk 2]
```diff
@@ -422,7 +422,7 @@ def concat_prompt_completion(example):
                     map_kwargs["desc"] = f"Tokenizing {dataset_name} dataset"
 
                 def tokenize(example, processing_class, dataset_text_field):
-                    return processing_class(example[dataset_text_field])
+                    return processing_class(text=example[dataset_text_field])
```
Comment (kashif) [inline trl/trainer/sft_trainer.py:425]: ah yes good catch! i forgot this one

File: examples/scripts/sft_gemma3.py (hunk 1 above)
Comment (qgallouedec) [reply examples/scripts/sft_gemma3.py:34]: sdpa gives me OOM currently. Let's keep eager until sdpa is fixed?

Comment (kashif) [reply examples/scripts/sft_gemma3.py:34]: ok...

File: trl/trainer/sft_trainer.py (hunk 2 above)
Comment (kashif) [reply trl/trainer/sft_trainer.py:425]: yes I believe so... since the Gemma3 processor has images as the first argument and text as the 2nd and its being called without any arguments..

**Review by kashif — APPROVED** [review]
